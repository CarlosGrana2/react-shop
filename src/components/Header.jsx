import React, { useState, useContext } from "react";
import "@styles/Header.scss";
import MobileMenu from "@components/MobileMenu.jsx";
import Menu from "@components/Menu.jsx";
import MyOrder from "@containers/MyOrder.jsx";
import AppContext from "@context/AppContext";

import menu from "@icons/icon_menu.svg";
import logo from "@logos/logo_yard_sale.svg";
import shoppingCart from "@icons/icon_shopping_cart.svg";

const Header = () => {
  const [toggleDesktop, setToggleDesktop] = useState(false);
  const [toggleMobile, setToggleMobile] = useState(false);
  const [toggleOrders, setToggleOrders] = useState(false);
  const {
    state: { cart },
  } = useContext(AppContext);

  const handleClickDesktop = () => {
    setToggleDesktop(!toggleDesktop);
  };

  const handleClickMobile = () => {
    setToggleMobile(!toggleMobile);
  };

  const verifyCart = (cartNumber) => {
    if (cartNumber && cartNumber > 9) {
      return "+9";
    } else {
      return cartNumber;
    }
  };
  return (
    <nav>
      <img src={menu} alt="logo" className="menu" onClick={handleClickMobile} />
      <div className="navbar-left">
        <a href="/" className="nav-logo">
          <img src={logo} alt="logo" />
        </a>
        <ul>
          <li>
            <a href="/">All</a>
          </li>
          <li>
            <a href="/">Clothes</a>
          </li>
          <li>
            <a href="/">Electronics</a>
          </li>
          <li>
            <a href="/">Furnitures</a>
          </li>
          <li>
            <a href="/">Toys</a>
          </li>
          <li>
            <a href="/">Others</a>
          </li>
        </ul>
      </div>
      <div className="navbar-right">
        <ul>
          <li className="navbar-email" onClick={handleClickDesktop}>
            cgranadosmontenegro
          </li>
          <li
            className="navbar-shopping-cart"
            onClick={() => setToggleOrders(!toggleOrders)}
          >
            <img src={shoppingCart} alt="shopping cart" />
            {cart.length > 0 && <div>{verifyCart(cart.length)}</div>}
          </li>
        </ul>
      </div>
      {toggleDesktop && <Menu />}
      {toggleMobile && <MobileMenu />}
      {toggleOrders && (
        <MyOrder
          toggleOrders={toggleOrders}
          setToggleOrders={setToggleOrders}
        />
      )}
    </nav>
  );
};

export default Header;
