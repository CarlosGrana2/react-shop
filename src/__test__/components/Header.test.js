import React from "react";
import { mount, shallow } from "enzyme";
import { create } from "react-test-renderer";
import Header from "../../components/Header";
import AppContext from "../../context/AppContext";

describe("<Header />", () => {
  test("Render the component header", () => {
    const wrapper = shallow(
      <AppContext.Provider value={{ state: { cart: 0 } }}>
        <Header />
      </AppContext.Provider>
    );
    expect(wrapper.length).toEqual(1);
  });

  test("Render the Navbar", () => {
    const wrapper = mount(
      <AppContext.Provider value={{ state: { cart: 0 } }}>
        <Header />
      </AppContext.Provider>
    );
    expect(wrapper.find(".navbar-left").length).toEqual(1);
    expect(wrapper.find(".navbar-right").length).toEqual(1);
  });
});

describe("Header Snapshot", () => {
  test("Validate UI the Header component", () => {
    const header = create(
      <AppContext.Provider value={{ state: { cart: 0 } }}>
        <Header />
      </AppContext.Provider>
    );
    expect(header.toJSON()).toMatchSnapshot();
  });
});
