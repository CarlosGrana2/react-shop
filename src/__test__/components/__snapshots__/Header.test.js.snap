// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Snapshot Validate UI the Header component 1`] = `
<nav>
  <img
    alt="logo"
    className="menu"
    onClick={[Function]}
    src={Object {}}
  />
  <div
    className="navbar-left"
  >
    <a
      className="nav-logo"
      href="/"
    >
      <img
        alt="logo"
        src={Object {}}
      />
    </a>
    <ul>
      <li>
        <a
          href="/"
        >
          All
        </a>
      </li>
      <li>
        <a
          href="/"
        >
          Clothes
        </a>
      </li>
      <li>
        <a
          href="/"
        >
          Electronics
        </a>
      </li>
      <li>
        <a
          href="/"
        >
          Furnitures
        </a>
      </li>
      <li>
        <a
          href="/"
        >
          Toys
        </a>
      </li>
      <li>
        <a
          href="/"
        >
          Others
        </a>
      </li>
    </ul>
  </div>
  <div
    className="navbar-right"
  >
    <ul>
      <li
        className="navbar-email"
        onClick={[Function]}
      >
        cgranadosmontenegro
      </li>
      <li
        className="navbar-shopping-cart"
        onClick={[Function]}
      >
        <img
          alt="shopping cart"
          src={Object {}}
        />
      </li>
    </ul>
  </div>
</nav>
`;
